### PR TITLE
Todo refactor

### PR DESCRIPTION
**Pull request by Just-Jojo**
Refactors todo and adds some more customizable settings. I have tested this so that's cool >.>

### CODE REVIEW

File: todo/todo.py [hunk 1]
```diff
@@ -28,151 +28,141 @@
...
-            ctx=ctx,
-            item="Detailed creation/removing",
-            key="detailed_pop",
-            toggle=toggle,
-        )
+        """Have a more detailed adding, removing, and completing message"""
+        if toggle:
+            ed = "enabled"
+        else:
+            ed = "disabled"
```
Comment (Kreusada) [inline todo/todo.py:100]: We can probably merge this into a single line, as seen in the `usemd` function (L110). 

```suggestion
        ed = "enabled" if toggle else "disabled"
```

File: todo/todo.py [hunk 2]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
-                completed,
-                "Completed Todo List",
-            )
-        else:
-            await ctx.send("You don't have any completed todos yet!")
+        msg = "Added that as a todo!"
+        details = await self.config.user(ctx.author).detailed_pop()
+        if details:
+            msg += f"\n'{discord.utils.escape_markdown(todo)}'"
+        await ctx.send(content=msg)
```
Comment (Kreusada) [inline todo/todo.py:212]: No need to define content here (optional).

```suggestion
        await ctx.send(msg)
```

File: todo/todo.py [hunk 3]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
+        await ctx.send(
+            "WARNING, this will remove **ALL** of your todos. Would you like to remove your todos? (y/n)"
+        )
+        pred = MessagePredicate.yes_or_no(ctx)
+        try:
+            await self.bot.wait_for("message", check=pred)
+        except asyncio.TimeoutError:
+            await ctx.send("Okay, I won't delete your todos")
+        else:
+            if pred.result is False:
```
Comment (Kreusada) [inline todo/todo.py:275]: We can simplify this as `result` returns a bool.

```suggestion
            if not pred.result:
```

File: todo/todo.py [hunk 4]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
-
-    @todo.command()
-    async def sort(self, ctx: commands.Context):
-        """Sort your todos alphabetically"""
-        if not len(await self.config.user(ctx.author).todos()):
-            return await ctx.send("You don't have any todos!")
-        msg = await ctx.send("Would you like to sort your todos in reverse?")
+            todos = await self._number_lists(todos)
+            if await self.config.user(ctx.author).combined_lists():
+                if (c := await self.config.user(ctx.author).completed()) :
```
Comment (Kreusada) [inline todo/todo.py:290]: Not familiar with the walrus operator, eitherway there's no need for this space here, unless its for the walrus operator (which I doubt).

```suggestion
                if (c := await self.config.user(ctx.author).completed()):
```

File: todo/todo.py [hunk 5]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
+            await self.page_logic(ctx, completed, "Completed todos")
+
+    @complete.command(name="remove", aliases=["del", "rm"])
+    async def complete_remove(self, ctx, *indexes: positive_int):
+        """Remove compeleted todos"""
+        if not indexes:
+            return await ctx.send_help(ctx.command)
+        indexes = [x - 1 for x in indexes]
+        _ = await self.config.user(ctx.author).completed()
+        if not _:
```
Comment (Kreusada) [inline todo/todo.py:365]: Yes, you used to tell me off for the way I named my variables, so I'm doing it back 😄
Perhaps change this to something better from the code's perspective.

```suggestion
        completed = await self.config.user(ctx.author).completed()
        if not completed:
```

File: todo/todo.py [hunk 6]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
-        """Explain what a todo is and why this cog exists!"""
-        if comic:
-            if await ctx.embed_requested():
-                embed = discord.Embed(
-                    title="About ToDo", colour=await ctx.embed_colour()
-                )
-                embed.set_footer(text="Jojo's ToDo Cog")
-                embed.set_image(url=_comic_link)
-                kwargs = {"embed": embed}
+            if pred.result is False:
```
Comment (Kreusada) [inline todo/todo.py:399]: We can simplify this as `result` will only return a bool.

```suggestion
            if not pred.result:
```

File: todo/todo.py [hunk 7]
```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
...
-        else:
-            await self.maybe_send_embed(
-                ctx=ctx, message=_about, title="About ToDo", footer="Jojo's Todo Cog"
-            )
+                await self.config.user(ctx.author).completed.clear()
+                await ctx.send("Removed your completed todos.")
+
+    ### Utility methods ###
+
+    async def page_logic(self, ctx: commands.Context, data: list, title: str):
```
Comment (Kreusada) [inline todo/todo.py:407]: For your utility functions, seeing as this cog is not documented through toctree, is it necessary for these expansive docstrings? Completely optional - but unneeded imo.

**Review by Kreusada — COMMENTED** [review]
Overall it's looking very good. A few suggested (mostly optional) changes in the review which you can commit or ignore to your liking, nothing breaking as such.

**Comment by Just-Jojo** [comment]
Thanks for the suggestions!

**Pull request merged**